src/rpdk/core/contract/resource_client.py

```diff
@@ -268,17 +268,17 @@ def convert_type(document_input, transformed_property):
     def compare(document_input, document_output):
         try:
             if isinstance(document_input, str):
```

Comment:
I am curious if this check is enough?

like if it is `str` then do regex match, otherwise just compare `==`. Should we have something more strict? what other alternatives there could be?

Reply:
As far as I know, regex match is only going to happen in case of a string in any other primitive type the value should be compared with ```===``` as we will have the exact value from the function.

Reply:
I could see number inputs needing `propertyTransform` in addition to string inputs

Reply:
I think I didn't explain well, let me take another stab at it. I understand that propertyTransform can be used for all primitive types namely: bool, string, number etc.
My point above was if a regex string is formed for any primitive type. The output for```if isinstance(document_input, str):```  where, document_input is regex will always be true. Therefore in case of any primitive type if a regex is formed we will execute the regex match. Does this make more sense?